pkg/apis/openebs.io/scp/v1alpha1/storage-volume.go

```diff
@@ -216,100 +100,99 @@ type Affinity struct {
 	TopologySpreadConstraint []v1.TopologySpreadConstraint `json:"topologySpreadConstraint,omitempty"`
 }
 
+// VolumeAffinityTerm specifies affinity requirements for a StorageVolume
 type VolumeAffinityTerm struct {
+	// TopologyKey is the key of cohort labels. StorageCohort that have a label with this key
+	// and identical values are considered to be in the same topology.
+	// +optional
 	TopologyKey string `json:"topologyKey,omitempty"`
 
+	// A label query over a set of resources, in this case storage volume.
+	// +optional
 	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
 }
 
 // StorageVolumeStatus defines the observed state of StorageVolume
-// +k8s:openapi-gen=true
 type StorageVolumeStatus struct {
 	// Phase defines the state of the volume
-	Phase StorageVolumePhase `json:"phase"`
+	Condition []StorageVolumeCondition `json:"condition,omitempty"`
 
 	// Capacity stores total, used and available size
-	// +nullable
+	// +optional
 	Capacity Capacity `json:"capacity,omitempty"`
 
-	// NVMESpec defines the nvme spec i.e nql, targetIP and status
-	// +nullable
-	NVMESpec []NVMESpec `json:"NVMESpec,omitempty"`
+	// TargetInfo defines the nvme spec i.e nql, targetIP and status
+	// +optional
+	TargetInfo []TargetInfo `json:"targetInfo,omitempty"`
+}
+
+// StorageVolumeCondition contains condition information for a StorageVolume.
+type StorageVolumeCondition struct {
+	// Type is the type of the condition.
+	Type StorageVolumeConditionType `json:"type"`
+
+	// Status is the status of the condition.
+	// Can be True, False, Unknown.
+	Status v1.ConditionStatus `json:"status"`
+
+	// Last time the condition transitioned from one status to another.
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+
+	// Last time we got an update on a given condition.
+	// +optional
+	LastHeartbeatTime metav1.Time `json:"lastHeartbeatTime,omitempty"`
+
+	// Unique, one-word, CamelCase reason for the condition's last transition.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// Human-readable message indicating details about last transition.
+	// +optional
+	Message string `json:"message,omitempty"`
 }
 
+// StorageVolumeConditionType is a valid value for StorageVolumeCondition.Type
+type StorageVolumeConditionType string
+
+const (
+	// StorageVolumeConditionTypePending indicates pending status of the StorageVolume object
+	StorageVolumeConditionTypePending StorageVolumeConditionType = "Pending"
+
+	// StorageVolumeConditionTypeScheduled represents scheduled StorageVolume object
+	StorageVolumeConditionTypeScheduled StorageVolumeConditionType = "Scheduled"
+
+	// StorageVolumeConditionTypeReady represents StorageVolume object is in ready state
+	StorageVolumeConditionTypeReady StorageVolumeConditionType = "Ready"
```

Comment:
One for error as well.
And thinking of having one for Updating (can be discussed).